src/XrdTls/XrdTlsSocket.cc

```diff
@@ -352,6 +353,14 @@ int XrdTlsSocket::Diagnose(const char *what, int sslrc, int tcode)
                           "TLS error rc=%d ec=%d (%s) errno=%d.",
                           sslrc, eCode, XrdTls::ssl2Text(eCode), eNO);
                  XrdTls::Emsg(pImpl->traceID, eBuff, true);
+                 // Report any additional queued errors
+                 char eBuffSSL[120]; int eQueue;
+                 while ((eQueue = ERR_get_error())) {
+                     ERR_error_string_n(eQueue, eBuffSSL, sizeof(eBuffSSL));
+                     snprintf(eBuff, sizeof(eBuff),
+                              "TLS error queue: ec=%d, %s", eQueue, eBuffSSL);
+                     XrdTls::Emsg(pImpl->traceID, eBuff, true);
+                 }
```

Comment:
This whole section is not needed. The call to XrdTls::Emsg(0 does this alreay.

Reply:
Great, I didn't realize Emsg handled that. I had set `xrd.trace all`, but hadn't noticed log or stderr entries for the error queue entries.